src/script/components/LexicalInput/LexicalInput.tsx

```diff
@@ -0,0 +1,177 @@
+  setInputValue: (text: string) => void;
+  editMessage: (messageEntity: ContentMessage, editor: LexicalEditor) => void;
+  children: ReactElement;
+  hasLocalEphemeralTimer: boolean;
+  saveDraftState: (editor: string) => void;
+  loadDraftState: () => Promise<DraftState>;
+  mentionCandidates: User[];
+  onShiftTab: () => void;
+}
+
+export const LexicalInput = forwardRef<LexicalEditor, LexicalInputProps>(
+  (
+    {
+      placeholder,
+      propertiesRepository,
+      searchRepository,
+      inputValue,
+      setInputValue,
+      children,
+      hasLocalEphemeralTimer,
+      saveDraftState,
+      loadDraftState,
+      editMessage,
+      mentionCandidates,
+      onShiftTab,
+    },
+    ref,
+  ) => {
+    // Emojis
+    const [shouldReplaceEmoji, setShouldReplaceEmoji] = useState<boolean>(
+      propertiesRepository.getPreference(PROPERTIES_TYPE.EMOJI.REPLACE_INLINE),
+    );
+
+    const editorConfig: InitialConfigType = {
+      namespace: 'WireLexicalEditor',
+      theme,
+      onError(error: unknown) {
+        // eslint-disable-next-line no-console
+        console.log('[LexicalInput.tsx] przemvs error', error);
+        throw error;
+      },
+      nodes: [BeautifulMentionNode, BeautifulMentionNode],
+    };
+
+    const queryMentions = (queryString?: string | null) => {
+      return queryString ? searchRepository.searchUserInSet(queryString, mentionCandidates) : mentionCandidates;
+    };
+
+    useEffect(() => {
+      amplify.subscribe(WebAppEvents.PROPERTIES.UPDATE.EMOJI.REPLACE_INLINE, setShouldReplaceEmoji);
+      amplify.subscribe(WebAppEvents.PROPERTIES.UPDATED, (properties: WebappProperties) => {
+        setShouldReplaceEmoji(properties.settings.emoji.replace_inline);
+      });
+    }, []);
+
+    return (
+      <LexicalComposer initialConfig={editorConfig}>
+        <div className="controls-center">
+          <div css={{width: '100%'}} className={cx('input-bar--wrapper')}>
+            <AutoFocusPlugin />
+            <GlobalEventsPlugin onShiftTab={onShiftTab} />
+            <EditorRefPlugin editorRef={ref} />
+            <DraftStatePlugin setInputValue={setInputValue} loadDraftState={loadDraftState} />
+            <EditMessage onMessageEdit={editMessage} />
+
+            {shouldReplaceEmoji && <EmojiPickerPlugin />}
+
+            <PlainTextPlugin
+              contentEditable={
+                <ContentEditable
+                  value={inputValue}
+                  className="conversation-input-bar-text"
+                  data-uie-name="input-message"
+                />
+              }
+              placeholder={<Placeholder text={placeholder} hasLocalEphemeralTimer={hasLocalEphemeralTimer} />}
+              ErrorBoundary={LexicalErrorBoundary}
+            />
+
+            <BeautifulMentionsPlugin onSearch={queryMentions} />
+
+            <OnChangePlugin
+              onChange={(editorState, lexicalEditor) => {
+                lexicalEditor.registerTextContentListener(textContent => {
```

Comment:
useCallback to prevent unnecessary function re-creations on each render.

Reply:
why are we creating function inside render, we should create the function outside render  to prevent unnecessary function re-creations on each render.